src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -1340,6 +1340,12 @@ public static function destroy($ids)
 
         $ids = is_array($ids) ? $ids : func_get_args();
 
+        foreach ($ids as $i => $id) {
+            if ($id instanceof Model) {
+                $ids[$i] = $id->getKey();
+            }
+        }
+
```

Comment:
better approach:
```php
foreach ($ids as &$id) {
    if ($id instanceof Model) {
        $id = $id->getKey();
    }
}
```

Reply:
why using the references is better?

Reply:
cleaner syntax... you don't need to keep track of the iterator key `$i`